### PR TITLE
feat(decorator): add `@flattenProperty`

### LINKED ISSUE #13: [Feature Request] Add `flattenProperty` to TCGC
Add a decorator in TCGC like :

```
@@flattenProperty(Foo.properties, [scope]);
```

And expose this property in SDK type as a boolean, let codegen decide how to handle it.

### PR DESCRIPTION
**Pull request by archerzz**
- implement `@flattenProperty`
- implement `shouldFlattenProperty` utility function
- add test cases

resolve #13

### CODE REVIEW

File: packages/typespec-client-generator-core/test/decorators.test.ts [hunk 1]
```diff
@@ -2197,4 +2206,102 @@ describe("typespec-client-generator-core: decorators", () => {
...
+          model Model2{}
+
+          @test
+          @route("/func1")
+          op func1(@body body: Model1): void;
+        }
+      `)) as { Model1: Model };
+
+      const childProperty = Model1.properties.get("child");
+      notEqual(childProperty, undefined);
```
Comment (timotheeguerin) [inline packages/typespec-client-generator-core/test/decorators.test.ts:2255]:
```suggestion
      strictNotEqual(childProperty, undefined);
```
or use the new `vitest` `expect`

File: packages/typespec-client-generator-core/test/decorators.test.ts [hunk 2]
```diff
@@ -2197,4 +2206,102 @@ describe("typespec-client-generator-core: decorators", () => {
       strictEqual(getUsage(runner.context, Dog), UsageFlags.Output);
     });
   });
+
+  describe("@flattenProperty", () => {
```
Comment (timotheeguerin) [inline packages/typespec-client-generator-core/test/decorators.test.ts:2210]: Should you add validation to make sure you don't create conflict?

File: packages/typespec-client-generator-core/test/decorators.test.ts (hunk 1 above)
Comment (archerzz) [reply packages/typespec-client-generator-core/test/decorators.test.ts]: replaced with `notStrictEqual`

File: packages/typespec-client-generator-core/test/decorators.test.ts (hunk 2 above)
Comment (timotheeguerin) [reply packages/typespec-client-generator-core/test/decorators.test.ts:2210]: talked offline, a name will be made up like previously with swagger, this was the discussion of dpg teams

File: docs/libraries/typespec-client-generator-core/reference/decorators.md [hunk 3]
```diff
@@ -172,6 +172,34 @@ model ModelToExclude {
 }
 ```
 
+### `@flattenProperty` {#@Azure.ClientGenerator.Core.flattenProperty}
+
+Set whether a model property should be flattened or not.
+
+```typespec
```
Comment (haolingdong-msft) [inline docs/libraries/typespec-client-generator-core/reference/decorators.md:179]: Do we want to mention in the doc that this decorator is not recommended to use?

Comment (archerzz) [reply docs/libraries/typespec-client-generator-core/reference/decorators.md:179]: This doc is auto-generated. I actually have `#deprecated "@flattenProperty decorator is not recommended to use."` in the original code comment.

**Review by lirenhe — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**